src/clients/scopes/AddScopeDialog.tsx

```diff
@@ -22,24 +31,65 @@ import "./client-scopes.css";
 
 export type AddScopeDialogProps = {
   clientScopes: ClientScopeRepresentation[];
+  clientName?: string;
   open: boolean;
   toggleDialog: () => void;
   onAdd: (
-    scopes: { scope: ClientScopeRepresentation; type: ClientScopeType }[]
+    scopes: { scope: ClientScopeRepresentation; type?: ClientScopeType }[]
   ) => void;
+  isClientScopesConditionType?: boolean;
 };
 
+enum FilterType {
+  Name = "Name",
+  Protocol = "Protocol",
+}
+
+enum ProtocolType {
+  All = "All",
+  SAML = "SAML",
+  OpenIDConnect = "OpenID Connect",
+}
+
 export const AddScopeDialog = ({
   clientScopes,
+  clientName,
   open,
   toggleDialog,
   onAdd,
+  isClientScopesConditionType,
 }: AddScopeDialogProps) => {
   const { t } = useTranslation("clients");
   const [addToggle, setAddToggle] = useState(false);
   const [rows, setRows] = useState<ClientScopeRepresentation[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.Name);
+  const [protocolType, setProtocolType] = useState<ProtocolType>(
+    ProtocolType.All
+  );
+  const [key, setKey] = useState(0);
+  const refresh = () => setKey(key + 1);
+
+  const [isFilterTypeDropdownOpen, setIsFilterTypeDropdownOpen] =
+    useState(false);
 
-  const loader = () => Promise.resolve(clientScopes);
+  const [isProtocolTypeDropdownOpen, setIsProtocolTypeDropdownOpen] =
+    useState(false);
+
+  useEffect(() => {
+    refresh();
+  }, [filterType, protocolType]);
+
+  const loader = () => {
+    if (protocolType === "OpenID Connect") {
+      return Promise.resolve(
+        clientScopes.filter((item) => item.protocol === "openid-connect")
+      );
+    } else if (protocolType === "SAML") {
+      return Promise.resolve(
+        clientScopes.filter((item) => item.protocol === "saml")
+      );
+    } else return Promise.resolve(clientScopes);
+  };
```

Comment:
If you make this function `async` there is no need to wrap the returns in a `Promise.resolve` call, which can reduce the code here. Also since we already have enums with string values we can compare against the enum instead.
```suggestion
  const loader = async () => {
    if (protocolType === ProtocolType.OpenIDConnect) {
      return clientScopes.filter((item) => item.protocol === "openid-connect");
    } else if (protocolType === ProtocolType.SAML) {
      return clientScopes.filter((item) => item.protocol === "saml");
    }

    return clientScopes;
  };
```